core/lib/server/stages.ts

```diff
@@ -172,8 +172,5 @@ export const movePub = (pubId: PubsId, stageId: StagesId, trx = db) => {
 			.with("leave_stage", (db) => db.deleteFrom("PubsInStages").where("pubId", "=", pubId))
 			.insertInto("PubsInStages")
 			.values([{ pubId, stageId }])
-			// Without this on conflict clause, the db errors if this function is called with the
-			// stageId the pub already belongs to
-			.onConflict((oc) => oc.doNothing())
```

Comment:
This was the cause of the stage being removed when you saved a pub. The onConflict handling here was allowing the `leave_stage` cte to execute successfully, and the insert to quietly do nothing, which resulted in removing the pub's stage.

Reply:
lol.